### PR TITLE
Rep4304

### PR DESCRIPTION
**Pull request by JoelRizner**
This should be released as 7.3.7.0 i believe.

### CODE REVIEW

**Comment by adrianjgeorge** [comment]
you need to update the poms to be 7.3.7.0-SNAPSHOT

**Comment by adrianjgeorge** [comment]
Looks fine, hoping it proves unneccessary

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 1]
```diff
@@ -152,7 +154,9 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
       "clusterId" -> clusterId,
       "nodeId" -> nodeId,
       "requestorIp" -> Option(stripHeaderParams(httpServletRequest.getHeader(CommonHttpHeader.X_FORWARDED_FOR.toString)))
-        .getOrElse(httpServletRequest.getRemoteAddr)
+        .getOrElse(httpServletRequest.getRemoteAddr),
+      "userResp" -> Option(stripHeaderParams(httpServletResponse.getHeader(X_PP_USER))).orNull,
+      "groupsResp" -> Option(stripHeaderParams(httpServletResponse.getHeader(X_PP_GROUPS))).orNull
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: Kind of odd. You shouldn't need to wrap anything in an `Option` here, if null is an acceptable value.

That is, if `stripHeaderParams(...)` returns null, well, that's what you want.

File: repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala [hunk 2]
```diff
@@ -54,6 +54,8 @@ class HerpFilter @Inject()(configurationService: ConfigurationService,
   private final val DEFAULT_CONFIG = "highly-efficient-record-processor.cfg.xml"
   private final val X_PROJECT_ID = "X-Project-ID"
   private final val X_METHOD_LABEL: String = "X-METHOD-LABEL"
+  private final val X_PP_USER: String = "X-PP-User"
+  private final val X_PP_GROUPS: String = "X-PP-Groups"
```
Comment (dmnjohns) [inline repose-aggregator/components/filters/herp/src/main/scala/org/openrepose/filters/herp/HerpFilter.scala]: Considering this is a `final val` (i.e., a constant) of an obvious type (i.e., `String`), I would say that annotating the type is not really necessary. That's just personal preference, though.

**Comment by adrianjgeorge** [comment]
This work has been superseded.

**Pull request closed without merging**